gadwords.php

```diff
@@ -19,140 +19,114 @@
  * needs please refer to http://www.prestashop.com for more information.
  *
  *  @author    PrestaShop SA <contact@prestashop.com>
- *  @copyright 2007-2015 PrestaShop SA
+ *  @copyright 2007-2019 PrestaShop SA
  *  @version	Release: $Revision: 17142 $
  *  @license   http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
  *  International Registered Trademark & Property of PrestaShop SA
  */
 
-if (!defined('_PS_VERSION_'))
+if (!defined('_PS_VERSION_')) {
 	exit;
+}
+
+require_once dirname(__FILE__).'/classes/GAdwordsModuleManagement.php';
 
 class GAdwords extends Module
 {
+	public $name;
+	public $tab;
+	public $version;
+	public $author;
+	public $bootstrap;
+	public $displayName;
+	public $description;
+	public $ps_versions_compliancy;
+	public $prestashopAdsModuleName;
+	public $prestashopAdsModuleId;
+	public $isPrestashop16;
 
 	public function __construct()
 	{
 		$this->name = 'gadwords';
 		$this->tab = 'advertising_marketing';
-		$this->version = '1.3.6';
+		$this->version = '2.0.0';
 		$this->author = 'PrestaShop';
-		$this->need_instance = 1;
-
 		$this->bootstrap = true;
 		parent::__construct();
 
 		$this->displayName = $this->l('Google AdWords');
 		$this->description = $this->l('You want to be more visible on Google and attract new clients ? Use our 75€ promo code on Google Adwords !');
-
-		if (_PS_VERSION_ < '1.5')
-			require(_PS_MODULE_DIR_.$this->name.'/backward_compatibility/backward.php');
-
-		if (!isset($this->_path))
-			$this->_path = _PS_MODULE_DIR_.$this->name;
+		$this->ps_versions_compliancy = array('min' => '1.6', 'max' => _PS_VERSION_);
+		$this->prestashopAdsModuleName = 'emarketing';
+		$this->prestashopAdsModuleId = 18716;
```

Comment:
Are `$prestashopAdsModuleName` & `$prestashopAdsModuleId` allowed to change during the code execution ? If no, you can define them as class constants.

Reply:
Thanks for the help.